src/main/java/Duke.java

```diff
@@ -1,10 +1,92 @@
+// Assuming other class definitions remain the same
+
 public class Duke {
+    private static final String LINE_SEPARATOR = "____________________________________________________________";
+    private static final List<Task> tasks = new ArrayList<>();
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner scanner = new Scanner(System.in);
+
+        printWelcomeMessage();
+
+        boolean isRunning = true;
+        while (isRunning) {
+            String input = scanner.nextLine();
+            printLineSeparator();
+
+            if ("bye".equalsIgnoreCase(input)) {
+                isRunning = false;
+            } else if ("list".equalsIgnoreCase(input)) {
+                listTasks();
+            } else {
+                handleTaskInput(input);
+            }
+
+            printLineSeparator();
+        }
+
+        printGoodbyeMessage();
+        scanner.close();
+    }
+
+    private static void printWelcomeMessage() {
+        printLineSeparator();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        printLineSeparator();
+    }
+
+    private static void printGoodbyeMessage() {
```

Comment:
Good extraction of methods to avoid having many print calls in your methods. Looks neater this way!